velox/exec/OutputBuffer.h

```diff
@@ -124,10 +157,45 @@ class DestinationBuffer {
   // The sequence number of the first item to pass to 'notify'.
   int64_t notifySequence_{0};
   uint64_t notifyMaxBytes_{0};
+  BufferStats stats_;
 };
 
 class Task;
 
+/// The stats of an OutputBuffer.
+struct OutputBufferStats {
+  /// BufferState indicates the OutputBuffer's current state.
+  enum class BufferState {
+    /// The OutputBuffer is open and ready to receive data. Initial state.
+    kOpen,
+    /// The OutputBuffer would not accept more buffers.
+    kNoMoreBuffers,
+    /// The OutputBuffer has received all input data, and some data is still
+    /// to be sent.
+    kFlushing,
+    /// The OutputBuffer has already received and sent all the input data.
+    kFinished
+  };
+
+  OutputBufferStats(
+      core::PartitionedOutputNode::Kind k,
+      bool noMoreBuffers,
+      bool atEnd,
+      bool finished,
+      const std::vector<BufferStats>& bufs);
+
+  /// The kind of the OutputBuffer.
+  core::PartitionedOutputNode::Kind kind;
```

Comment:
make all of these const
revise comment to remove redundant text

Reply:
done.